airflow/hooks/hive_hooks.py

```diff
@@ -487,22 +487,22 @@ def get_results(self, hql, schema='default', arraysize=1000):
                 'data': [],
                 'header': [],
             }
+            cur = conn.cursor()
```

Comment:
The dbapi does not mandate it, but it is implemented in the impala API, see here (https://github.com/cloudera/impyla/blob/master/impala/hiveserver2.py#L190)

I really think it is the best way to do this, especially because the close would attempt to kill the operation in progress on HiveServer2.